src/main/java/io/jenkins/plugins/coverage/adapter/XMLCoverageReportAdapter.java

```diff
@@ -36,22 +36,27 @@ public XMLCoverageReportAdapter(String path) {
      * @param source source xml file
      */
     @Override
-    public Document convert(File source) {
+    public Document convert(File source) throws ConversionException {
         File xsl;
         try {
             xsl = getRealXSL();
         } catch (FileNotFoundException e) {
             e.printStackTrace();
             throw new ConversionException(e);
         }
-        return XMLUtils.getInstance().convertToDocumentWithXSL(xsl, source);
+        try {
+            return XMLUtils.getInstance().convertToDocumentWithXSL(xsl, source);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            throw new ConversionException(e);
+        }
     }
 
     @SuppressWarnings("WeakerAccess")
-    protected File getRealXSL() throws FileNotFoundException {
+    protected File getRealXSL() throws ConversionException, FileNotFoundException {
         try {
             File realXSL = new File(getClass().getResource(getXSL()).toURI());
-            if(!realXSL.exists()) {
+            if (!realXSL.exists()) {
```

Comment:
It would also make sense to check whether the `File` is actually a file (e.g. it may be pointing to a directory). there is `IsFile()` method